Source/SwiftLintFramework/Rules/GenericTypeNameRule.swift

```diff
@@ -0,0 +1,226 @@
+            }
+        }
+
+        let contents = file.contents.bridge()
+        return namesAndRanges.flatMap { (name, range) -> (String, Int)? in
+            guard let byteRange = contents.NSRangeToByteRange(start: range.location + offset,
+                                                              length: range.length),
+                case let kinds = file.syntaxMap.tokensIn(byteRange).flatMap({ SyntaxKind(rawValue: $0.type) }),
+                kinds == [.identifier] else {
+                    return nil
+            }
+
+            return (name, byteRange.location)
+        }
+    }
+
+    private func validateName(name: String, file: File, offset: Int) -> [StyleViolation] {
+        guard !configuration.excluded.contains(name) else {
+            return []
+        }
+
+        if !CharacterSet.alphanumerics.isSuperset(ofCharactersIn: name) {
+            return [
+                StyleViolation(ruleDescription: type(of: self).description,
+                               severity: .error,
+                               location: Location(file: file, byteOffset: offset),
+                               reason: "Generic type name should only contain alphanumeric characters: '\(name)'")
+            ]
+        } else if !name.substring(to: name.index(after: name.startIndex)).isUppercase() {
+            return [
+                StyleViolation(ruleDescription: type(of: self).description,
+                               severity: .error,
+                               location: Location(file: file, byteOffset: offset),
+                               reason: "Generic type name should start with an uppercase character: '\(name)'")
+            ]
+        } else if let severity = severity(forLength: name.characters.count) {
+            return [
+                StyleViolation(ruleDescription: type(of: self).description,
+                               severity: severity,
+                               location: Location(file: file, byteOffset: offset),
+                               reason: "Generic type name should be between \(configuration.minLengthThreshold) and " +
+                    "\(configuration.maxLengthThreshold) characters long: '\(name)'")
+            ]
+        }
+
+        return []
+    }
+}
+
+extension String {
+    fileprivate func split(separator: String) -> [(String, NSRange)] {
+        let separatorLength = separator.bridge().length
+        var previousEndOffset = 0
+        var result = [(String, NSRange)]()
+
+        for component in components(separatedBy: separator) {
+            let length = component.bridge().length
+            let range = NSRange(location: previousEndOffset, length: length)
+            result.append((component, range))
+            previousEndOffset += length + separatorLength
+        }
+
+        return result
+    }
+
+    fileprivate func trimmingWhitespaces() -> (String, NSRange) {
+        let range = NSRange(location: 0, length: bridge().length)
```

Comment:
since you call `bridge()` twice in this function, could you instead just save it to a local `let bridged = bridge()` variable?